Add dependency to docker/swarmkit

aluzzardi opened the pull request:
Add a dependency to docker/swarmkit in preparation for adding first-class Swarm-wide service management inside the Docker Engine as described in our [ROADMAP.md](https://github.com/docker/docker/blob/master/ROADMAP.md#14-cluster-capable-engine).

tonistiigi committed:
Add dependency to docker/swarmkit

Add a dependency to `docker/swarmkit` in preparation for adding
first-class Swarm-wide service management inside the Docker Engine as
described in our ROADMAP.md.

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>
Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>
Signed-off-by: Victor Vieux <vieux@docker.com>
Signed-off-by: Madhu Venugopal <madhu@docker.com>

sfsmithcha committed:
add tutorial

Signed-off-by: Charles Smith <charles.smith@docker.com>

ibuildthecloud commented on hack/vendor.sh:
```diff
@@ -59,14 +59,14 @@ clone git github.com/vdemeester/shakers 24d7f1d6a71aa5d9cbe7390e4afb66b7eef9e1b3
-clone git github.com/docker/go-connections v0.2.0
-clone git github.com/docker/engine-api 772250a752e34cacaeef7c92b8e0ddf43450b629
+clone git github.com/docker/go-connections fa2850ff103453a9ad190da0df0af134f0314b3d
+clone git github.com/docker/engine-api 6e7fae3cd9aa40e31cb5833d67a94ee8cff3054c git://github.com/vieux/engine-api.git
```
I assume this won't stay as vieux's repo?

thaJeztah replied:
@ibuildthecloud it should be updated after https://github.com/docker/engine-api/pull/265 is accepted and merged

thaJeztah replied:
Thanks for asking though!

vdemeester replied:
@ibuildthecloud definitely, there is a PR in review for that in `engine-api` :angel:

vdemeester replied:
damm @thaJeztah is quick :joy_cat:

justincormack commented:
This needs a rebase.

crosbymichael commented:
If we get the engine-api change merged this one can be merged easily.  Its just a bunch of vendored deps

tonistiigi committed:
Add Swarm management backend

As described in our ROADMAP.md, introduce new Swarm management API
endpoints relying on swarmkit to deploy services. It currently vendors
docker/engine-api changes.

This PR is fully backward compatible (joining a Swarm is an optional
feature of the Engine, and existing commands are not impacted).

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>
Signed-off-by: Victor Vieux <vieux@docker.com>
Signed-off-by: Daniel Nephin <dnephin@docker.com>
Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>
Signed-off-by: Madhu Venugopal <madhu@docker.com>

tonistiigi committed:
Swarm integration tests

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>
Signed-off-by: Victor Vieux <vieux@docker.com>

Commit:
Add swarm api reference docs

Signed-off-by: Arnaud Porterie (icecrime) <arnaud.porterie@docker.com>

dnephin committed:
Add Swarm management CLI commands

As described in our ROADMAP.md, introduce new Swarm management commands
to call to the corresponding API endpoints.

This PR is fully backward compatible (joining a Swarm is an optional
feature of the Engine, and existing commands are not impacted).

Signed-off-by: Daniel Nephin <dnephin@docker.com>
Signed-off-by: Victor Vieux <vieux@docker.com>
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

amitshukla committed:
CLI docs

Signed-off-by: Amit Shukla <amit.shukla@docker.com>
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "swarm" git@github.com:docker/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~95
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

icecrime commented:
In the making of this PR:
- CLI was merged into this branch from #23363 (complete history is still available on this PR)
- API and backend implementation was merged into this branch from #23362 (complete history is still available on this PR)

This branch is now updated with a squashed version of this work to avoid polluting master's history.

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "swarm" git@github.com:docker/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~8
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

icecrime commented:
LGTM :+1:

crosbymichael commented:
LGTM

vieux commented:
LGTM

icecrime commented:
Ping @SvenDowideit: can you please help us with that?

```
1:04:59 ERROR: 2016/06/14 04:04:59 site.go:1211: Two or more menu items have the same name/identifier in Menu "main": "update".
21:04:59 Rename or set an unique identifier.
```

icecrime commented:
Nevermind we figured it out :innocent:

icecrime commented:
![image](https://cloud.githubusercontent.com/assets/1564054/16031193/51b01d88-31ad-11e6-81bc-b67ececd640d.png)

vieux commented:
@mlaventure 
![bumble_bee](https://cloud.githubusercontent.com/assets/1032519/16031333/d14a4540-31ae-11e6-9b5c-2a5c4370326a.png)

tiborvass commented:
![](https://ascent.atos.net/wp-content/uploads/2015/09/atos-ascent-swarm-computing-what-goes-around-comes-around.-2.jpg)

tonistiigi commented:
LGTM

tiborvass commented:
LGTM

vdemeester commented:
experimal timed out after all the tests 😇
